result.go

```diff
@@ -273,33 +272,44 @@ type flusher interface {
 	Flush()
 }
 
+// Encode will encode the results into the writer using the Encoder and separating each entry
```

Comment:
This comment or a similar one should also precede the interface method for `MultiResultEncoder` in this file.

Reply:
My thoughts were that it would be up to the implementation for what they did so I left the interface ambiguous. Is that alright?

Reply:
I guess it's okay if it's left up to the implementation, but in that case it would be good to note that in the interface comment.

Reply:
Done.